### ISSUE TITLE
Incompatible with Python 3.10

### ISSUE THREAD

**Issue by bertonha** [jazzband/django-push-notifications#622]
This package is not supported on Python 3.10.

Mainly because of the sub dependency `hyper` (discontinued) is importing `Iterable` from `collection` instead of `collection.abc`

`hyper` is dependency of `apns2`

**Comment by koedal** [comment]
I also have this issue. Are there any plans to support Python 3.10?

**Comment by andrewsanchez** [comment]
Quick fix is to patch it like so:


https://github.com/healthvana/h2/commit/d67c6ca10eb7f79c0737c37fdecfe651307a7414

**Comment by samirsd** [comment]
can we address this? it's been 3 months now. is there a plan to merge this patch or otherwise?

**Comment by samirsd** [comment]
> Quick fix is to patch it like so:
> 
> [healthvana/h2@d67c6ca](https://github.com/healthvana/h2/commit/d67c6ca10eb7f79c0737c37fdecfe651307a7414)

would you mind explaining how to do this? I'm assuming you'd start by forking `django-push-notifications` and whatever dependency is problematic and use that...? however I'm not sure where to find the `hyper` or `apns2` packages in this project.

**Comment by merwok** [comment]
No, that link shows a monkey-patch, meaning changing a module at runtime.  It is not a source patch to be applied to source code.

**Comment by shree-ranga** [comment]
Is this not fixed yet? I'm still facing module not found 'apns2'.

**Comment by azmeuk** [comment]
Closing in favor of #675

**Comment by harikvpy** [comment]
+1

**Comment by koedal** [comment]
Any updates here?

**Comment by ethanitovitch** [comment]
Still seeing this issue on python 3.11

**Comment by kmasuhr** [comment]
+1

**Comment by mirodil1** [comment]
+1

**Comment by ssyberg** [comment]
Still an issue!

**Comment by SaravananRamanathan** [comment]
I see that, this was opened on `Oct 6, 2021`, and it's still a issue until now, anyone know of the current state of things? what do we need to support python 3.10+ ?

**Comment by qx** [comment]
still bug! Is there anyone still maintaining this project?

**Comment by ssyberg** [comment]
I believe this is addressed in this PR that I think is almost ready for merge: https://github.com/jazzband/django-push-notifications/pull/696

**Comment by qx** [comment]
@ssyberg you can test by yourself, It easy to appear

**Comment by ssyberg** [comment]
> @ssyberg you can test by yourself, It easy to appear

I don't know what you mean

**Comment by qx** [comment]
This is a prone issue and I don't think it's been fixed yet

**Comment by ssyberg** [comment]
> This is a prone issue and I don't think it's been fixed yet

You mean you don't think that PR addresses it? As I mentioned it hasn't been merged yet.

**Comment by coratype** [comment]
So has this been fixed? I'm getting the Iterable thing

**Comment by illymarev** [comment]
As Python 3.8 is close to EOL, many people are migrating to newer versions. Are there any updates on this issue?

**Comment by DataGreed** [comment]
Any updates on this? Having the samy issue with hyper on python 3.11

**Comment by DataGreed** [comment]
> Quick fix is to patch it like so:
> 
> [healthvana/h2@d67c6ca](https://github.com/healthvana/h2/commit/d67c6ca10eb7f79c0737c37fdecfe651307a7414)

Thanks, it helped as a temporary workaround. Hope for this lib to be updated for 3.10+ though :)
